app/_src/mesh/features/provenance-verification-images.md

```diff
@@ -0,0 +1,165 @@
+---
+title: Verify Build Provenance for Signed Kong Mesh Images
+badge: enterprise
+---
+
+Starting with 2.7.4, {{site.mesh_product_name}} produces build provenance for docker container images, which can be verified using `cosign` / `slsa-verifier` with attestations published to a Docker Hub repository.
+
+This guide provides steps to verify build provenance for signed {{site.mesh_product_name}} Docker container images in two different ways:
+
+* A minimal example, used to verify an image without leveraging any annotations
+* A complete example, leveraging optional annotations for increased trust
+
+For the minimal example, you only need a Docker manifest digest and a GitHub repo name.
+
+{:.important .no-icon}
+> The Docker manifest digest is required for build provenance verification. The manifest digest can be different from the platform specific image digest for a specific distribution.
+
+For the complete example, you need the same details as the minimal example, as well as any of the optional annotations you wish to verify:
+
+| Shorthand | Description | Example Value |
+|---|---|---|
+| `<repo>` | GitHub repository | `kong-mesh` |
+| `<workflow name>` | GitHub workflow name | `build-test-distribute` |
+| `<workflow trigger>` | Github workflow trigger name | `push` |
+| `<version>` | Artifact version to download | `2.7.4` |
+
+Because Kong uses GitHub Actions to build and release, Kong also uses GitHub's OIDC identity to generate build provenance for container images, which is why many of these details are GitHub-related.
+
+## Examples
+
+### Prerequisites
+
+For both examples, you need to:
+
+1. Ensure `cosign` / `slsa-verifier` is installed.
+
+2. Ensure `regctl` is installed.
+
+3. Collect the necessary image details.
+
+4. Parse the `<manifest_digest>` for the image using `regctl`.
+
+   ```sh
+   regctl manifest digest <image>:<tag>
+   ```
+
+5. Set the `COSIGN_REPOSITORY` environment variable:
+
+   ```sh
+   export COSIGN_REPOSITORY=kong/notary
+   ```
+
+{:.important .no-icon}
+> The GitHub owner is case-sensitive (`Kong/kong-mesh` vs `kong/kong-mesh`).
+
+### Minimal example
+
+#### Using Cosign
+
+Run the `cosign verify-attestation ...` command:
+
+```sh
+cosign verify-attestation \
+   <image>:<tag>@sha256:<manifest_digest> \
+   --type='slsaprovenance' \
+   --certificate-oidc-issuer='https://token.actions.githubusercontent.com' \
+   --certificate-identity-regexp='^https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@refs/tags/v[0-9]+.[0-9]+.[0-9]+$'
+```
+
+Here's the same example using sample values instead of placeholders:
+
+```sh
+cosign verify-attestation \
+   'kong/kuma-cp:2.7.4@sha256:<manifest_digest>' \
```

Comment:
replace <manifest_digest> with image digest for official 2.7.4 tag. Provenance is NOT generated for preview tags to test

Reply:
@lahabana  the `cosign` verification method works with just using the `<image>:<tag>` where <tag> can be automated  / made dynamic using the `{{page.latest_version}}`. 

However the `slsa-verifier` method **requires (otherwise the command fails)** the `<image>:<tag>@sha256:<image_digest>` in which case, if i automate the <tag>, we would still need a parse the <image_digest> which is statically populated for an example tag. 

We have 2 options:
1. **Use only** the `Cosign` cli tool and **Remove** the `slsa-verifier` cli tool to verify the image  provenance. This approach works fine and can be automated. 
   - Customer examples will use only single tool to verify everything
   - Can be automated

2. **Use the static tag and sha** and **ignore** the dynamic example values across release branches. This is what happens on GW pages currently. 
   - Examples with both tools
   - Cannot be automated for `slsa-verifier` that requires `<image_digest>`